zebra-state/src/service.rs

```diff
@@ -843,13 +841,23 @@ impl Service<Request> for ReadStateService {
         Poll::Ready(Ok(()))
     }
 
-    #[instrument(name = "read_state", skip(self, req))]
+    #[instrument(name = "read_state", skip(self))]
     fn call(&mut self, req: Request) -> Self::Future {
         match req {
-            // TODO: implement for lightwalletd before using this state in RPC methods
-
             // Used by get_block RPC.
-            Request::Block(_hash_or_height) => unimplemented!("ReadStateService doesn't Block yet"),
+            Request::Block(hash_or_height) => {
+                let state = self.clone();
+
+                async move {
+                    Ok(read::block(
+                        state.best_chain_receiver.borrow().as_ref(),
```

Comment:
I think this will hold the lock while the `read::block` executes. Would it make sense to clone the `Arc` here instead?

```suggestion
                        state.best_chain_receiver.borrow().clone().as_ref(),
```

Reply:
Fixed in d98981 by adding a wrapper that always clones the borrow.

I did this for all of the state watch channels, so we never block on reading the watched data.